feat(settings/advanced): custrom piston retraction

Angel-Padilla opened the pull request:
## What was done?

The option `initial retraction` from  the `brew settings` menu changed to `shot volume`, with it You can set amount of water for the shots to come

The volume range goes from 100ml to 150ml. That value is purely visual and used to calculate the retraction distance (mm) for the piston to be saved in the settings menu. Calculation is done using the cylinder diameter of 53 mm

![shot volume](https://github.com/user-attachments/assets/1bd87d76-c51e-4e52-9ed2-fee8e6ce9722)

> **Note: requieres approval of [this PR](https://github.com/MeticulousHome/meticulous-backend/pull/270) to succesfully save float values in the `partial_retraction` setting**

Angel-Padilla committed:
feat(settings/advanced): custrom piston retraction

A new option 'Shot volume' is added to the advanced settings menu, with
it You can set the retraction distance the piston follows to let it
the selected volume of water to the cylinder.

The volume range goes from 100ml to 150ml. That value is purely visual and
used to calculate the retraction distance (mm) for the piston to be saved
in the settings menu.

Mimoja commented:
We already have this in the brew menu. Please use that instead

Angel-Padilla committed:
chore(routes): rename shot volume gauge

Angel-Padilla commented:
Oversaw it, do You mind if I stick to the `shot volume` name? as I think is more intuitive for users

Mimoja commented:
Sure. Feel free to rename. I decided against because not all machines use the same kinematic

Angel-Padilla committed:
chore(brewsettings): relocate shot volume option

The shot volume option is moved from advanced settings to brew settnigs
where it was a version of it already implemented

AluUriel commented:
:tada: This PR is included in version 1.90.0 :tada:

The release is available on [GitHub release](https://github.com/MeticulousHome/meticulous-dial/releases/tag/v1.90.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
